Tests/Rules/UseProcessBlockForPipelineCommand.tests.ps1

```diff
@@ -0,0 +1,37 @@
+﻿Describe "UseProcessBlockForPipelineCommand" {
+    BeforeAll {
+        $RuleName = 'PSUseProcessBlockForPipelineCommand'
+        $NoProcessBlock = 'function BadFunc1 { [CmdletBinding()] param ([Parameter(ValueFromPipeline)]$Param1) }'
+        $NoProcessBlockByPropertyName = 'function $BadFunc2 { [CmdletBinding()] param ([Parameter(ValueFromPipelineByPropertyName)]$Param1) }'
+        $HasProcessBlock = 'function GoodFunc1 { [CmdletBinding()] param ([Parameter(ValueFromPipeline)]$Param1) process { } }'
+        $HasProcessBlockByPropertyName = 'function GoodFunc2 { [CmdletBinding()] param ([Parameter(ValueFromPipelineByPropertyName)]$Param1) process { } }'
+        $NoAttribute = 'function GoodFunc3 { [CmdletBinding()] param ([Parameter()]$Param1) }'
+		$HasTypeDeclaration = 'function GoodFunc3 { [CmdletBinding()] param ([Parameter()][string]$Param1) }'
+    }
+
+    Context "When there are violations" {
+        $Cases = @(
+            @{ScriptDefinition = $NoProcessBlock; Name = "NoProcessBlock"}
+            @{ScriptDefinition = $NoProcessBlockByPropertyName; Name = "NoProcessBlockByPropertyName"}
+        )
+        It "has 1 violation for function <Name>" {
+            param ($ScriptDefinition)
+
+            Invoke-ScriptAnalyzer -ScriptDefinition $ScriptDefinition -IncludeRule $RuleName | Should -Not -BeNullOrEmpty
+        } -TestCases $Cases
+    }
+
+    Context "When there are no violations" {
+        $Cases = @(
```

Comment:
Please move this definition to either a `BeforeAll` or to be a literal value in the `TestCases` parameter

Reply:
fixed in #707f02d96ac5763afaf0b69649d787f8f25b2492

Moved the script definitions to the cases array.

Reply:
Also took the opportunity to give the tests more descriptive names.